src/adapter/src/coord.rs

```diff
@@ -1737,6 +1737,15 @@ impl Coordinator {
         let write_frontier = self.least_valid_write(&id_bundle);
         // Things go wrong if we try to create a dataflow with `as_of = []`, so avoid that.
         if write_frontier.is_empty() {
+            tracing::info!(
+                export_ids = dataflow.format_export_ids(),
+                %cluster_id,
+                min_as_of = ?min_as_of.elements(),
+                write_frontier = ?write_frontier.elements(),
+                "selecting index `as_of` as {:?}",
```

Comment:
Should we specifically call out in the message that we are using the read frontier because the write frontier was empty?

Also, what happens here if the read frontier is also empty?

Reply:
> Should we specifically call out in the message that we are using the read frontier because the write frontier was empty?

I thought about that, but decided against it as I don't think log messages are a good place to explain how our as_of selection logic works. I think when you use these logs to debug production issues, you will also have a copy of the code open and then the logic should be obvious from the comments here.

> Also, what happens here if the read frontier is also empty?

Good question! We will return the empty frontier as an `as_of` here, so "things go wrong". But they would go wrong regardless of what we return because an empty read frontier means that the inputs cannot be read at any time, so that dataflow could never come up healthy.